MiscVBAFunctions/Modules/MiscTables.bas

```diff
@@ -191,3 +191,142 @@ Function TableColumnToArray(TableDicts As Collection, ColumnName As String) As V
+    '
+    ' Returns:
+    '   Collection of the selected column.
+    
+    Dim col1 As Collection
+    Dim dict As Dictionary
+    
+    Set col1 = New Collection
+    For Each dict In TableDicts
+        col1.Add dictget(dict, ColumnName)
+    Next dict
+    
+    Set TableColumnToCollection = col1
+End Function
+
+
+Public Sub ResizeLO(SelectedLO As ListObject, NumRows As Long)
+    ' Resize a table to the desired number of data rows. The columns remain unchanged.
+    ' If NumRows is set to "0", the table will instead be resized to "1" and the
+    ' content in that 1st row will be cleared.
+    '
+    ' Args:
+    '   SelectedLO: Selected List Object.
+    '   NumRows: Number of desired rows.
+    
+    Dim oldNumRows As Long
+    oldNumRows = SelectedLO.ListRows.Count
+    
+    ' Don't allow tables to be resized to zero rows.  Excel won't really do
+    ' this anyway - it's possible to get the DataBodyRange to be empty, in
+    ' which case the single row displayed in the table is the Insert row, but
+    ' this creates more problems than it solves.  Instead, resize the table to
+    ' one row, and set a flag to clear out any remaining data later.
+    Dim clearTable As Boolean
+    clearTable = False
+    If NumRows = 0 Then
+        NumRows = 1
+        clearTable = True
+    End If
+    
+    ' Resize the table (add 1 to the number of rows because mListObject.Range
+    ' includes the header row).
+    SelectedLO.Resize _
+        SelectedLO.Range.Resize( _
+            NumRows + 1, _
+            SelectedLO.ListColumns.Count)
+    
+    ' If the table is resized to have one row, but the row contains no data,
+    ' the row will be treated as the Insert row, and the data row count will
+    ' remain zero.  This will cause problems since the table doesn't actually
+    ' have a DataBodyRange.  To avoid this situation, put a space in the first
+    ' column, which will cause the Insert row to change to a data row.  After
+    ' setting the value once, it can be removed and the row will remain part
+    ' of the DataBodyRange.
+    
+    If NumRows = 1 And SelectedLO.ListRows.Count = 0 Then
+        SelectedLO.Range.Cells(1, 1).Offset(1, 0).Value = " "
+        SelectedLO.DataBodyRange.ClearContents
+    End If
+    
+    ' The user requested that the table be resized to zero rows.  We resized
+    ' it down to one row, now clear out the data.
+    
+    If clearTable Then
+        SelectedLO.DataBodyRange.ClearContents
+        'ClearSort
+    End If
+    
+    ' If the new number of rows is less than the old number of rows, clear out
+    ' the rows that were just removed from the table.
+    If NumRows < oldNumRows Then
+        SelectedLO.DataBodyRange _
+            .Offset(NumRows, 0) _
+            .Resize(oldNumRows - NumRows, SelectedLO.ListColumns.Count) _
+            .ClearContents
+    End If
+End Sub
+
+
+Public Function GetColumnLO(SelectedLO As ListObject, ColumnName As String) As Range
```

Comment:
I would rename `SelectedLO` to just `LO`. But maybe @rudolfbyker can add his two cents here as the "naampolisie". 

Also not sure about the function name as I would interpret it to return the `ListColumn` object, ie  `LO.ListColumns(ColumnName)`. Maybe this is a better signature (no need to specify LO in the function name as the first argument has a `ListObject` type):
```
Function GetTableColumnRange(
    LO as ListObject,
    ColumnName as String
)
```